File: app/models/organization_member.rb
# frozen_string_literal: true

class OrganizationMember
  attr_reader :id, :name, :email, :role

  def initialize(id:, name:, email:, role:)
    @id = id
    @name = name
    @email = email
    @role = role
  end

  def as_json()
    {
      id: id,
      name: name,
      email: email,
      role: role
    }
  end
end

File: lib/tuist_cloud_web/api/schemas/organization_member.ex
defmodule TuistCloudWeb.API.Schemas.OrganizationMember do
  @moduledoc """
  A schema for an organization member.
  """
  require OpenApiSpex
  alias OpenApiSpex.Schema

  OpenApiSpex.schema(%{
    type: :object,
    description: "An organization member",
    properties: %{
      id: %Schema{
        type: :number,
        description: "The organization member's unique identifier"
      },
      email: %Schema{
        type: :string,
        description: "The organization member's email"
      },
      name: %Schema{
        type: :string,
        description: "The organization member's name"
      },
      role: %Schema{
        type: :string,
        enum: ["admin", "user"],
        description: "The organization member's role"
      }
    },
    required: [:id, :email, :name, :role]
  })
end
